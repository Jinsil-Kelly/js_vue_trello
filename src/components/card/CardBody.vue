<template>
  <div>
    <h3>Description</h3>
    <textarea
      class="form-control"
      cols="30"
      rows="3"
      placeholder="Add a more detailed description..."
      :readonly="!toggleDesc"
      @click="$emit('onToggleDesc')"
      @blur="$emit('onBlurDesc', $event.target.value)"
      :value="description"
    ></textarea>
  </div>
</template>

<script>
export default {
  name: "EditCardHeader",
  props: ["description", "toggleDesc"]
};
</script>

<style scoped></style>
