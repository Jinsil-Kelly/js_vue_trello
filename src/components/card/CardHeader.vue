<template>
  <div class="modal-card-header">
    <div class="modal-card-header-title">
      <input
        class="form-control"
        :readonly="!toggleTitle"
        @click="$emit('onToggleTitle')"
        @blur="$emit('onBlurTitle', $event.target.value)"
        type="text"
        :value="title"
      />
    </div>
    <a class="modal-close-btn" href="" @click.prevent="$emit('close')"
      >&times;</a
    >
  </div>
</template>

<script>
export default {
  name: "EditCardHeader",
  props: ["title", "toggleTitle"]
};
</script>

<style scoped></style>
