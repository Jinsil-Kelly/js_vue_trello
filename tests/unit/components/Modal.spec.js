//https://alexjover.com/blog/test-vue-js-slots-in-jest/

import { mount } from "@vue/test-utils";
import Modal from "@/components/Modal";
import CardBody from "@/components/card/CardBody.vue";
import CardHeader from "@/components/card/CardHeader.vue";
describe("Modal", () => {
  let cmp;
  beforeEach(() => {
    const cardHeaderWrapper = {
      render(h) {
        return h(CardHeader, { props: { title: "card1" } });
      }
    };

    const cardBodyWrapper = {
      render(h) {
        return h(CardBody, { props: { title: "card1" } });
      }
    };

    cmp = mount(Modal, {
      slots: {
        header: cardHeaderWrapper,
        body: cardBodyWrapper,
        footer: "<div/>"
      }
    });
  });

  test("EditCardHeader are inserted in a Modal component", () => {
    expect(cmp.find(CardHeader).isVueInstance()).toBe(true);
    expect(cmp.find(CardBody).isVueInstance()).toBe(true);
    expect(cmp.html()).toMatchSnapshot();
  });

  test("Header slot renders a default header text", () => {
    const header = cmp.find(".modal-header");
    expect(header.find("input").element.value).toBe("card1");
  });

  test("Body slot renders h3 and textarea text", () => {
    const body = cmp.find(".modal-body");
    expect(body.find("h3").text()).toBe("Description");
    expect(body.find("textarea").exists()).toBe(true);
  });

  test("Footer slot doesn't have any child El", () => {
    const footer = cmp.find(".modal-footer");
    expect(footer.html()).toEqual(
      '<div class="modal-footer"><div></div></div>'
    );
  });
});
