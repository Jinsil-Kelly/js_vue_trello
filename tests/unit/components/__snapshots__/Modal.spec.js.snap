// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal EditCardHeader are inserted in a Modal component 1`] = `
<div class="modal-mask" name="modal">
  <div class="modal-wrapper">
    <div class="modal-container">
      <div class="modal-header">
        <div class="modal-card-header">
          <div class="modal-card-header-title"><input readonly="readonly" type="text" class="form-control"></div> <a href="" class="modal-close-btn">Ã—</a>
        </div>
      </div>
      <div class="modal-body">
        <div>
          <h3>Description</h3> <textarea cols="30" rows="3" placeholder="Add a more detailed description..." readonly="readonly" class="form-control"></textarea>
        </div>
      </div>
      <div class="modal-footer">
        <div></div>
      </div>
    </div>
  </div>
</div>
`;
